frontend/src/features/properties/list/PropertyListView.test.tsx

```diff
@@ -165,6 +167,104 @@ describe('Property list view', () => {
+      const { getByTestId } = render(
+        <Provider store={store}>
+          <Router history={history}>
+            <PropertyListView />
+          </Router>
+        </Provider>,
+      );
+      expect(getByTestId('edit-icon')).toBeInTheDocument();
+      fireEvent(
+        getByTestId('edit-icon'),
+        new MouseEvent('click', { bubbles: true, cancelable: true }),
+      );
+      await wait(() => expect(getByTestId('save-changes')).toBeInTheDocument(), { timeout: 500 });
+    });
+  });
+
+  it('Enables edit on property rows that the user has the same agency as the property', async () => {
+    setupTests([{ ...mockFlatProperty }]);
+
+    await act(async () => {
+      const { getByTestId, container } = render(
+        <Provider store={store}>
+          <Router history={history}>
+            <PropertyListView />
+          </Router>
+        </Provider>,
+      );
+      expect(getByTestId('edit-icon')).toBeInTheDocument();
+      fireEvent(
+        getByTestId('edit-icon'),
+        new MouseEvent('click', { bubbles: true, cancelable: true }),
+      );
+      await wait(
+        () => {
+          expect(getByTestId('save-changes')).toBeInTheDocument();
+          expect(container.querySelector(`input[name="properties.0.market"]`)).not.toBeNull();
+        },
+        { timeout: 500 },
+      );
+    });
+  });
+
+  it('Disables property rows that the user does not have edit permissions for', async () => {
+    setupTests([{ ...mockFlatProperty, agencyId: 2 }]);
+
+    await act(async () => {
+      const { getByTestId, container } = render(
+        <Provider store={store}>
+          <Router history={history}>
+            <PropertyListView />
+          </Router>
+        </Provider>,
+      );
+      expect(getByTestId('edit-icon')).toBeInTheDocument();
+      fireEvent(
+        getByTestId('edit-icon'),
+        new MouseEvent('click', { bubbles: true, cancelable: true }),
+      );
+      await wait(
+        () => {
+          expect(getByTestId('save-changes')).toBeInTheDocument();
+          expect(container.querySelector(`input[name="properties.0.market"]`)).toBeNull();
+        },
+        { timeout: 500 },
+      );
+    });
+  });
+
+  it('Disables property rows that are in an active project', async () => {
+    setupTests([{ ...mockFlatProperty, projectNumbers: ['SPP-10000'] }]);
+
+    await act(async () => {
+      const { container, getByTestId } = render(
+        <Provider store={store}>
+          <Router history={history}>
+            <PropertyListView />
+          </Router>
+        </Provider>,
+      );
+      expect(getByTestId('edit-icon')).toBeInTheDocument();
+      fireEvent(
+        getByTestId('edit-icon'),
+        new MouseEvent('click', { bubbles: true, cancelable: true }),
+      );
+      await wait(
+        () => {
+          expect(getByTestId('save-changes')).toBeInTheDocument();
+          expect(container.querySelector(`input[name="properties.0.market"]`)).toBeNull();
```

Comment:
I think this is good enough, but I'd like to circle back and explicitly check for the read only version of the row.